tests/integration-tests/tests/common/data/system-analyzer.sh

```diff
@@ -0,0 +1,169 @@
+#!/bin/bash -eua
+#
+# Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Requirements: sudo, user be able to become root without password, ethtool, netstat
+
+# Script to generate a system information archive.
+# This script needs to be run by a user in sudoers and it takes an existing directory as argument
+
+function body () {
+  IFS= read -r header
+  printf '%s\n' "${header}"
+  "$@"
+}
+
+function sighdl ()  {
+  echo "catching a signal!"
+    if [ "$1" != "0" ]; then
+      rm -fr "$BASE_TEMP_DIR"
```

Comment:
here you're using an environment variable defined later, let's add a comment or let's pass it as parameter.

Then let's use `${VARIABLE_NAME}` syntax everywhere, for each bash variable.

Reply:
Done